Week 04/Lecture 07/Assignment 02/lecture_7/src/main/java/com/example/lecture_7/config/AppConfig.java

```diff
@@ -0,0 +1,9 @@
+package com.example.lecture_7.config;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ComponentScan(basePackages = "com.example.lecture_7")
```

Comment:
When we need this ComponentScan (notice the anotation SpringBootApplication). Do we really need @ComponentScan, what is the case ?

Reply:
Thank you for the feedback.

The `@SpringBootApplication` annotation is a convenience annotation that combines three annotations:
- `@Configuration`: Designates the class as a configuration class.
- `@EnableAutoConfiguration`: Enables Spring Boot’s auto-configuration mechanism.
- `@ComponentScan`: Enables component scanning so that the web controller classes and other classes you create will be automatically discovered and registered as Spring beans.

Since `@SpringBootApplication` already includes `@ComponentScan,` i generally do not need to explicitly declare `@ComponentScan` again, unless i need to customize the component scanning, such as specifying a different base package, which is not.

I have removed the redundant `@ComponentScan` annotation from AppConfig to avoid unnecessary duplication. The updated code is pushed in the latest commit.